config/util.go

```diff
@@ -0,0 +1,23 @@
+package config
+
+import "bytes"
+
+func unquoteTomlString(b []byte) []byte {
+	if len(b) >= 6 {
+		if bytes.HasPrefix(b, []byte(`'''`)) && bytes.HasSuffix(b, []byte(`'''`)) {
+			return b[3 : len(b)-3]
+		}
+		if bytes.HasPrefix(b, []byte(`"""`)) && bytes.HasSuffix(b, []byte(`"""`)) {
+			return b[3 : len(b)-3]
+		}
+	}
+	if len(b) >= 2 {
+		if bytes.HasPrefix(b, []byte(`'`)) && bytes.HasSuffix(b, []byte(`'`)) {
+			return b[1 : len(b)-1]
+		}
+		if bytes.HasPrefix(b, []byte(`"`)) && bytes.HasSuffix(b, []byte(`"`)) {
+			return b[1 : len(b)-1]
+		}
+	}
```

Comment:
Thanks for your review @jdstrand! Here are my answers to your points: ;-)

> `"""'''foo'''"""` returns `'''foo'''` whereas `'''"""foo"""'''` returns foo. Perhaps don't worry about `"""'''` or `'''"""` and use an `else if`?
> similar for "'foo'" and '"foo"'. Maybe use an else if?

The current behavior is that `"""'''foo'''"""` returns `'''foo'''` and `'''"""foo"""'''` returns `"""foo"""`, which is what I would expect. Anyway, I added those examples to the [unit-test](https://github.com/influxdata/telegraf/blob/9f7730ea6bd3cbd3d924c1f8c3e1a654132681c7/config/secret_test.go#L170). Likewise for mixed single quotes.
We don't need `else if`s in the `unquoteTomlString()` as we directly `return` on the first matching condition, so everything else is only executed if nothing matched previously...

> """"foo"""" returns foo. Perhaps the outer conditional should be else if too?

I tested and both quadruple single-quotes as well as double-quotes will trigger a TOML parsing error. Do you want me to add a test for this?

> It would be good to have some table driven tests for `unquoteTomlString()`
That's what `TestSecretUnquote()` in `secret_test.go` does in combination with the TOML parser, which will catch some of the odd cases. Do you want me to add another test for the underlying `unquoteTomlString()` function?

Reply:
Oh! I missed that each of those 'if's had return statements. Whoops, hehehe :) `secret_test.go` is fine. Thanks for adding those.